samza-core/src/main/java/org/apache/samza/table/utils/TableMetricsUtil.java

```diff
@@ -46,21 +43,16 @@
   /**
    * Constructor based on container context
    *
-   * @param containerContext container context
-   * @param taskContext task context
+   * @param context {@link Context} for this task
    * @param table underlying table
    * @param tableId table Id
    */
-  public TableMetricsUtil(SamzaContainerContext containerContext, TaskContext taskContext,
-      Table table, String tableId) {
-
-    Preconditions.checkNotNull(containerContext);
+  public TableMetricsUtil(Context context, Table table, String tableId) {
+    Preconditions.checkNotNull(context);
     Preconditions.checkNotNull(table);
     Preconditions.checkNotNull(tableId);
 
-    this.metricsRegistry = taskContext == null // The table is at container level, when the task
-        ? containerContext.metricsRegistry     // context passed in is null
-        : taskContext.getMetricsRegistry();
+    this.metricsRegistry = context.getTaskContext().getTaskMetricsRegistry();
```

Comment:
I was confused by this as well. I don't know how taskContext could be null. I tried searching through the code and couldn't find anything.
I don't think this would be a clean way to decide between task vs. container metrics anyways.
I found the change that added this check (https://github.com/apache/samza/pull/555), and I didn't see any cases where taskContext was ever null. @weisong44 could you please clarify why this check was here?

Reply:
We don't currently support container level table yet, once that 's added you'd see null taskContext.

Reply:
Thanks. We will simplify the constructor now to just use task-level metrics, since that is the only current use case. When the container-level table functionality comes in, then this can be modified or a new path can be added.